c10/core/TensorImpl.h

```diff
@@ -2422,57 +2526,108 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
+            is_channels_last_3d_ = bool(is_channels_last_3d);
+            is_non_overlapping_and_dense_ = bool(is_non_overlapping_and_dense);
+          }
+        };
 
-    is_contiguous_ = compute_contiguous();
+    auto is_contiguous = compute_contiguous();
     // Note:
     // Dim 0, 1, 2 will never be a channels last 2d/3d format
     // Dim 3+ is possibly be a channels last 2d format (Dim 4 only at this
     // point) Dim 4+ is possibly be a channels last 3d format (Dim 5 only at
     // this point)
     switch (dim()) {
-      case 4:
-        is_channels_last_contiguous_ = compute_channels_last_contiguous_2d();
-        is_channels_last_3d_contiguous_ = false;
-        is_channels_last_ = compute_strides_like_channels_last_2d();
-        is_channels_last_3d_ = false;
-        is_non_overlapping_and_dense_ = is_contiguous_ ||
-            is_channels_last_contiguous_ || compute_non_overlapping_and_dense();
+      case 4: {
+        auto is_channels_last_contiguous =
+            compute_channels_last_contiguous_2d();
+        set_fields(
+            is_contiguous,
+            is_channels_last_contiguous,
+            bool_is_channels_last_3d_contiguous(false),
+            compute_strides_like_channels_last_2d(),
+            bool_is_channels_last_3d(false),
+            bool_is_non_overlapping_and_dense(
+                is_contiguous || is_channels_last_contiguous ||
+                compute_non_overlapping_and_dense()));
         break;
-      case 5:
-        is_channels_last_contiguous_ = compute_channels_last_contiguous_2d();
-        is_channels_last_3d_contiguous_ = !is_channels_last_contiguous_ &&
-            compute_channels_last_contiguous_3d();
-        is_channels_last_ = !is_channels_last_3d_contiguous_ &&
-            compute_strides_like_channels_last_2d();
-        is_channels_last_3d_ =
-            !is_channels_last_ && compute_strides_like_channels_last_3d();
-        is_non_overlapping_and_dense_ = is_contiguous_ ||
-            is_channels_last_contiguous_ || is_channels_last_3d_contiguous_ ||
-            compute_non_overlapping_and_dense();
+      }
+      case 5: {
+        auto is_channels_last_contiguous =
+            compute_channels_last_contiguous_2d();
+        auto is_channels_last_3d_contiguous =
+            bool_is_channels_last_3d_contiguous(
+                !is_channels_last_contiguous &&
+                compute_channels_last_contiguous_3d());
+        auto is_channels_last = bool_is_channels_last(
+            !is_channels_last_3d_contiguous &&
+            compute_strides_like_channels_last_2d());
+        auto is_channels_last_3d = bool_is_channels_last_3d(
+            !is_channels_last && compute_strides_like_channels_last_3d());
+        auto is_non_overlapping_and_dense = bool_is_non_overlapping_and_dense(
+            is_contiguous || is_channels_last_contiguous ||
+            is_channels_last_3d_contiguous ||
+            compute_non_overlapping_and_dense());
+        set_fields(
```

Comment:
would these be more readable if you did

set_fields(
   bool_is_channels_last_contiguous_2d(
         compute_channels_last_contiguous_2d()),
   bool_is_channels_last_3d_contiguous(
         !is_channels_last_contiguous &&
         compute_channels_last_contiguous_3d()),
   ...

Reply:
No, because if I do it directly in set_fields I cannot reference the intermediate values

Reply:
oh yea.